Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/repository/storage/ValueSetsStorage.kt

```diff
@@ -26,72 +23,28 @@ class ValueSetsStorage @Inject constructor(
         context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
     }
 
-    var vaccinationValueSet: VaccinationValueSet
+    var valueSetsContainer: ValueSetsContainer
         get() = getValueSet()
         set(value) = setValueSet(value)
 
-    private fun getValueSet(): VaccinationValueSet {
-        Timber.v("Loading value set")
-        val valueSetString = prefs.getString(PKEY_VALUE_SETS_PREFIX, null)
+    private fun getValueSet(): ValueSetsContainer {
+        Timber.v("Loading value sets container")
+        val valueSetString = prefs.getString(PKEY_VALUE_SETS_CONTAINER_PREFIX, null)
         return when (valueSetString != null) {
-            true -> gson.fromJson(valueSetString, StoredVaccinationValueSet::class.java)
-            else -> emptyVaccinationValueSet
-        }.also { loaded -> Timber.v("Loaded value set %s", loaded) }
+            true -> gson.fromJson(valueSetString)
+            else -> emptyValueSetsContainer
+        }.also { loaded -> Timber.v("Loaded value sets container %s", loaded) }
     }
 
-    private fun setValueSet(value: VaccinationValueSet) {
-        Timber.v("Saving value set %s", value)
+    private fun setValueSet(value: ValueSetsContainer) {
+        Timber.v("Saving value sets container %s", value)
         prefs.edit {
-            val json = gson.toJson(value.toStoredVaccinationValueSet(), StoredVaccinationValueSet::class.java)
+            val json = gson.toJson(value, ValueSetsContainer::class.java)
```

Comment:
Do we need to to take care of old saved json ? like kind of migration ?

Reply:
migration is unnecessary because the value sets will be updated